src/odemis/gui/cont/stream.py

```diff
@@ -523,6 +529,10 @@ def _on_z_pos(self, zPos):
 
         self.stream.zIndex.subscribe(self._on_z_index)
 
+    def _on_max_projection(self, val):
```

Comment:
In theory, the VA might be changed via code which is not running in the main GUI thread. So to be safe, you have to decorate the function with `@call_in_wx_main`.

Note that the functions just above don't (directly) change the GUI, so they don't need to be protected.